doc/man3/EVP_MAC.pod

```diff
@@ -0,0 +1,235 @@
+may not be supported by the underlying implementation.
+
+EVP_MAC_update() adds C<datalen> bytes from C<data> to the MAC input.
+
+EVP_MAC_final() does the final computation and stores the result in
+the memory pointed at by C<out>, and sets its size in the B<size_t>
+the C<poutlen> points at.  If C<out> is B<NULL>, then no computation
+is made.  To figure out what the output length will be and allocate
+space for it dynamically, simply call with C<out> being B<NULL> and
+C<poutlen> pointing at a valid location, then allocate space and make
+a second call with C<out> pointing at the allocated space.
+
+EVP_MAC_ctrl() is used to give the MAC associated with the given
+context extra data that can't be easily given any other way.  This
+includes the MAC key, and for MACs that use other algorithms to do
+their computation, this is also the way to tell it which one to use.
+This functions takes variable arguments, the exact expected arguments
+depend on C<cmd>.  See </CONTROLS> below for a description of standard
+controls.
+
+EVP_MAC_ctrl_str() is an alternative to EVP_MAC_ctrl() to control the
+MAC implementation as E<lt> C<type>, C<value> E<gt> pairs.  The MAC
+implementation documentation should specify what control type strings
+are accepted.
+
+EVP_MAC_str2ctrl() and EVP_MAC_hex2ctrl() are helper functions to
+control the MAC implementation with raw strings or with strings
+containing hexadecimal numbers.  The latter are decoded into
+bitstrings that are sent on to EVP_MAC_ctrl().
+
+=head2 Information functions
+
+EVP_MAC_size() returns the MAC output size for the given context.
+
+EVP_MAC_nid() return the numeric identity of the given MAC implementation.
+
+EVP_MAC_name() return the numeric identity of the given MAC implementation.
+
+=head2 Registration functions
+
+EVP_add_mac() adds the MAC implementation C<mac> to the internal
+object database.
+
+EVP_get_macbyname() fetches a MAC implementation from the object
+database by name.
+
+EVP_get_macbynid() fetches a MAC implementation from the object
+database by numeric identity.
+
+EVP_get_macbyobj() fetches a MAC implementation from the object
+database by ASN.1 OBJECT (i.e. an encoded OID).
+
+=head1 CONTROLS
+
+The standard controls are:
+
+=over 4
+
+=item B<EVP_MAC_CTRL_SET_KEY>
+
+This control expects two arguments: C<unsigned char *key>, C<size_t keylen>
+
+These will set the MAC key from the given string of the given length.
+The string may be any bitstring, and can contain NUL bytes.
+
+For MACs that use an underlying computation algorithm, the algorithm
+I<must> be set first, see B<EVP_MAC_CTRL_SET_ENGINE>,
+B<EVP_MAC_CTRL_SET_MD> and B<EVP_MAC_CTRL_SET_CIPHER> below.
+
+=item B<EVP_MAC_CTRL_SET_FLAGS>
+
+This control expects one arguments: C<unsigned long flags>
+
+These will set the MAC flags to the given numbers.  Some MACs do not
+support this option.
+
+=item B<EVP_MAC_CTRL_SET_ENGINE>
+
+=item B<EVP_MAC_CTRL_SET_MD>
```

Comment:
The above 2 are duplicates documented below

Reply:
No.  These lines are items, the others are text within the block starting with these items.  You should look at the document with `perldoc` to see how it renders.

Reply:
You might want to use comma-separated items in a single "item" entry as was done in many other manpages.